x/core/keeper/data_request_indexing.go

```diff
@@ -39,14 +43,18 @@ func (k Keeper) GetDataRequestStatus(ctx sdk.Context, id string) (types.DataRequ
 }
 
 // GetDataRequestStatuses returns the statuses of the data requests given their IDs.
-func (k Keeper) GetDataRequestStatuses(ctx sdk.Context, ids []string) (map[string]types.DataRequestStatus, error) {
-	statuses := make(map[string]types.DataRequestStatus)
+func (k Keeper) GetDataRequestStatuses(ctx sdk.Context, ids []string) (map[string]*types.MaybeDataRequestStatus, error) {
```

Comment:
Don't think this has to be pointer: `*types.MaybeDataRequestStatus`

Reply:
It has to- the the method definition is proto generated.